app/service/file_svc.py

```diff
@@ -145,6 +155,27 @@ async def add_special_payload(self, name, func):
         """
         self.special_payloads[name] = func
 
+    def add_extension(self, ext, func):
+        """
```

Comment:
Should we combine `add_special_payload` and this function to clean them up?  I think we can *mostly* combine special payloads & extensions together